lib/request.js

```diff
@@ -102,6 +105,12 @@ function buildRequest (opts) {
       bodyTimeout: undiciOpts.bodyTimeout
     }
 
+    if (opts.url.protocol.indexOf('unix') === 0) {
+      done(new Error('unix socket not supported with undici yet'))
```

Comment:
This seems like a major obstacle to using it as the default. Would it not be better to solve this before switching this default?

Reply:
Maybe we can use undici as default and add a section in docs that points out that if `unix:` is needed `opts.undici` has to be set to false?

Reply:
I have added back most of the unix socket support I could add. I do not think supporting unix sockets as in `reply.from(path)` is useful. It will be supported as a base.